lib/parachain/types.go

```diff
@@ -0,0 +1,388 @@
+// Copyright 2023 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package parachain
+
+import (
+	"fmt"
+
+	"github.com/ChainSafe/gossamer/dot/types"
+
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/ChainSafe/gossamer/lib/crypto"
+	"github.com/ChainSafe/gossamer/lib/crypto/sr25519"
+	"github.com/ChainSafe/gossamer/pkg/scale"
+)
+
+// GroupRotationInfo A helper data-type for tracking validator-group rotations.
+type GroupRotationInfo struct {
+	// SessionStartBlock is the block number at which the session started
+	SessionStartBlock uint32 `scale:"1"`
+	// GroupRotationFrequency indicates how often groups rotate. 0 means never.
+	GroupRotationFrequency uint32 `scale:"2"`
+	// Now indicates the current block number.
+	Now uint32 `scale:"3"`
```

Comment:
I stated this above but gonna reiterate here, I'm a little concerned as to why we are representing it as a uint32 here. Is there a reason we are deviating from how we represent block number in the header? Worried about a situation where the block itself can process the number but other parts of the code cant

Reply:
Can we create a `BlockNumber` type alias in `dot/types/block.go` and start using it in every place?

Reply:
Ah, I had missed it there. Good catch, thank you!

Reply:
Resolving this to keep the conversation in one place. Feel free to open it if something isn't addressed